### DISCUSSION TITLE
ClearML support for train_pytorch

### DISCUSSION THREAD

**Discussion by hanshupe007** [mindee/doctr#1702] [General]
I saw that clearml is only supported in the train_tensorflow script, was there any reason why it's not implemented in train_pytorch?

**Comment by felixdittrich92** [comment]
Hi @hanshupe007 :wave:,
If I remember we had a problem with a memory leak.
But we have still an open ticket for it: https://github.com/mindee/doctr/issues/1458
CC @odulcy-mindee

**Comment by odulcy-mindee** [comment]
Hello ! Yes, a memory leak is observed when ClearML is used. Few PRs has been merged and IIRC one PR fixed an unclosed `open(...)` so maybe it has been fixed. I haven't tested for a while.

**Reply by hanshupe007** [reply]
Is there a version I can pull with ClearML support? because the current train_pytorch.py doesn't include any clearml code.

**Reply by felixT2K** [reply]
Hi @hanshupe007 :wave:,
Unfortunately no depends on @odulcy-mindee open ticket (#1458)

**Reply by pavelgein** [reply]
I found this issue while googling for known issues with Clearml, because we have encountered memory leak. A colleague of mine find the problem and opened the issue https://github.com/clearml/clearml/issues/1556

ClearML team says, that the Task should be inited as soon as possible, maybe this will be relevant to you
